File: TLBB1.5/Client/Assets/Client/Scripts/Table/Table_PowerPrice.cs
//This code create by CodeEngine mrd.cyou.com ,don't modify
using System;
 using System.Collections.Generic;
 using System.Collections;

namespace GCGame.Table{

[Serializable]
 public class Tab_PowerPrice : ITableOperate
{ private const string TAB_FILE_DATA = "PowerPrice.txt";
 public enum _ID
 {
 INVLAID_INDEX=-1,
ID_POWERPRICE,
MAX_RECORD
 }
 public string GetInstanceFile(){return TAB_FILE_DATA; }

private int m_PowerPrice;
 public int PowerPrice { get{ return m_PowerPrice;}}

public bool LoadTable(Hashtable _tab)
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),SerializableTable,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }
 public void SerializableTable(ArrayList valuesList,string skey,Hashtable _hash)
 {
 if (string.IsNullOrEmpty(skey))
 {
 throw TableException.ErrorReader("Read File{0} as key is Empty Fail!!!", GetInstanceFile());
 }

 if ((int)_ID.MAX_RECORD!=valuesList.Count)
 {
 throw TableException.ErrorReader("Load {0} error as CodeSize:{1} not Equal DataSize:{2}", GetInstanceFile(),_ID.MAX_RECORD,valuesList.Count);
 }
 Int32 nKey = Convert.ToInt32(skey);
 Tab_PowerPrice _values = new Tab_PowerPrice();
 _values.m_PowerPrice =  Convert.ToInt32(valuesList[(int)_ID.ID_POWERPRICE] as string);

 _hash.Add(nKey,_values); }


}
}

File: TLBB1.5/tabletools/CodeTable/Java/xjgame/table/Table_PowerPrice.java
//This code create by CodeEngine mrd.cyou.com ,don't modify
package xjgame.table;
import java.util.ArrayList;
 import java.util.HashMap;

public class Table_PowerPrice implements ITableOperate {
 public final static String TAB_FILE_DATA = "/PowerPrice.txt";
 public static final int ID_POWERPRICE = 0 ;
public static final int MAX_RECORD = 1 ;

 public String GetInstanceFile(){return TAB_FILE_DATA; }

private int m_PowerPrice;
 public int GetPowerPrice() { return m_PowerPrice;}

public boolean LoadTable(HashMap<Integer, Table_PowerPrice> _tab) throws TableException
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),this,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }

 @Override
 public void SerializableTable(ArrayList<String> valuesList,String skey,Object obj) throws TableException {
 @SuppressWarnings("unchecked")
 HashMap<Integer, Table_PowerPrice> _hash = (HashMap<Integer, Table_PowerPrice> )obj;
 if (skey.isEmpty())
 {
 throw TableException.ErrorReader("Read File"+GetInstanceFile()+" as key is Empty Fail!!!");
 }

 if (MAX_RECORD!=valuesList.size())
 {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as CodeSize:"+MAX_RECORD+" not Equal DataSize:"+valuesList.size());
 }
 try { 
 Integer nKey = Integer.parseInt(skey);
 Table_PowerPrice _values = new Table_PowerPrice();
 _values.m_PowerPrice =  Integer.parseInt((String)valuesList.get(ID_POWERPRICE));

 _hash.put(nKey,_values);
 } catch (NumberFormatException e) {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as :" + e.getMessage());
 }
 }


}

